tests/units/plugins/filter/test_merge.py

```diff
@@ -361,3 +362,210 @@ def test_services_merge_with_no_config(self):
+        }
+
+        self.assertEquals(
+            merge_with_database(base, database, "edxapp", "eugene", "development"),
+            expected
+        )
+
+    @mock.patch('plugins.filter.merge.random_password', return_value="foo")
+    # pylint: disable=unused-argument
+    def test_database_merge_with_same_database_and_different_release(self, random_password_mock):
+        """Test with an existing engine configured but not the same release"""
+
+        base = {
+            "postgresql": [
+                {
+                    "release": "9.6",
+                    "databases": [
+                        {
+                            "application": "edxapp",
+                            "name": "d_eugene_edxapp",
+                            "password": "bar",
+                            "user": "d_eugene_edxapp"
+                        }
+                    ]
+                }
+            ]
+        }
+        database = {"engine": "postgresql", "release": "10.0"}
+        expected = {
+            "postgresql": [
+                {
+                    "release": "9.6",
+                    "databases": [
+                        {
+                            "application": "edxapp",
+                            "name": "d_eugene_edxapp",
+                            "password": "bar",
+                            "user": "d_eugene_edxapp"
+                        }
+                    ]
+                },
+                {
+                    "release": "10.0",
+                    "databases": [
+                        {
+                            "application": "edxapp",
+                            "name": "d_eugene_edxapp",
+                            "password": "foo",
+                            "user": "d_eugene_edxapp"
+                        }
+                    ]
+                }
+            ]
+        }
+
+        self.assertEquals(
+            merge_with_database(base, database, "edxapp", "eugene", "development"),
+            expected
+        )
+
+    @mock.patch('plugins.filter.merge.random_password', return_value="foo")
+    # pylint: disable=unused-argument
+    def test_database_merge_with_other_environment(self, random_password_mock):
+        """Merge database in production environment"""
+
+        base = {}
+        database = {"engine": "postgresql", "release": "10.0"}
+        expected = {
+            "postgresql": [
+                {
+                    "release": "10.0",
+                    "databases": [
+                        {
+                            "application": "edxapp",
+                            "name": "p_eugene_edxapp",
+                            "password": "foo",
+                            "user": "p_eugene_edxapp"
+                        }
+                    ]
+                }
+            ]
+        }
+
+        self.assertEquals(
+            merge_with_database(base, database, "edxapp", "eugene", "production"),
+            expected
+        )
```

Comment:
Please test all other environments, not just `production`: we want to ensure that single letter codes match expected environments.